src/cluster/replication.cc

```diff
@@ -236,18 +237,22 @@ void ReplicationThread::CallbacksStateMachine::Start() {
     handlers_.emplace_front(CallbacksStateMachine::WRITE, "auth write", authWriteCB);
   }
 
+  uint64_t connect_timestamp = 0, connect_timeout_ms = 3100;
+
   while (!repl_->stop_flag_ && bev == nullptr) {
-    Status s = Util::SockConnect(repl_->host_, repl_->port_, &cfd);
+    if (Util::GetTimeStampMS() - connect_timestamp < 1000) {
```

Comment:
@caipengbo How about renaming connect_timestamp to last_connect_timestamp, and also adding a comment line to explain why we need this logic. 

// Sleep here to prevent frequent re-connect when the master is down with connection refused error

Reply:
Done